### PR TITLE
2040 setOrderNumber action

### PR DESCRIPTION
**Pull request by sdwvit**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,7 +6,6 @@ sphere-node-client-debug.log
 .idea
 lib-cov
 coverage.html
-coverage
```
Comment (pnedelko) [inline .gitignore]: why did you remove that?

File: lib/actions/setOrderNumber.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        .fetch()
+        .then(setOrderNumber)
+        .catch(handleError)
+        .finally(end)
+        .done();
+
+    function setOrderNumber(response) {
+        var order = response.body;
+        var orderNumber = data.orderNumber;
+        if (orderNumber === null || orderNumber === "") {
```
Comment (pnedelko) [inline lib/actions/setOrderNumber.js]: What if orderNumber is undefined? I'd propose you to do just `if (!orderNumber)`. It will cover all these cases.

File: lib/actions/setOrderNumber.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            throw new Error("Order number is required");
+        }
+        var orderId = data.orderId;
+        console.log('Executing setOrderNumber with ' + orderNumber + ' order number for order: ' + orderId);
```
Comment (pnedelko) [inline lib/actions/setOrderNumber.js]: To get rid from pluses: console.log('Executing setOrderNumber with %s order number for order: %s', orderNumber, orderId);

File: component.json [hunk 4]
```diff
@@ -224,6 +224,27 @@
                 }
             },
             "dynamicMetadata":true
+        },
+        "setOrderNumber": {
+            "main": "./lib/actions/setOrderNumber.js",
+            "title": "Set Order Number",
+            "metadata": {
+                "in": {
```
Comment (pnedelko) [inline component.json]: No "out" data here?

File: package.json [hunk 5]
```diff
@@ -13,7 +13,7 @@
     "node-uuid": "^1.4.1",
     "sphere-node-sdk": "^1.2.0",
     "moment": "2.0.0",
-    "underscore": "1.4.2"
+    "underscore": "^1.8.0"
```
Comment (pnedelko) [inline package.json]: You didn't use underscore in action but decided to update the version?

**Comment by pnedelko** [comment]
@sdwvit tests are required

File: .gitignore (hunk 1 above)
Comment (sdwvit) [inline .gitignore]: because it is mentioned twice in file

File: lib/actions/setOrderNumber.js (hunk 2 above)
Comment (sdwvit) [inline lib/actions/setOrderNumber.js]: tnx, will apply

File: package.json (hunk 5 above)
Comment (sdwvit) [inline package.json]: the component will not install without underscore v1.8.0

File: component.json (hunk 4 above)
Comment (sdwvit) [inline component.json]: yup

File: spec/actions/setOrderNumber.spec.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+                .reply(200, {
+                    version: 13
+                });
+
+            runs(function () {
+                setOrderNumber.process.call(self, msg, cfg);
+            });
+
+            waitsFor(function() {
+                return self.emit.calls.length;
```
Comment (pnedelko) [inline spec/actions/setOrderNumber.spec.js]: Better to use here `return self.emit.calls.length === 2;`
